migrate.py

```diff
@@ -1747,28 +1747,29 @@ def _rewrite_yaml_mapping_keys_non_vars(el, namespace, collection, spec, args):
             except LookupError:
                 pass
 
-        for ns in spec.keys():
-            for coll in get_rewritable_collections(ns, spec):
-                if collection == coll:
-                    # https://github.com/ansible-community/collection_migration/issues/156
-                    continue
-
-                try:
-                    modules_in_collection = get_plugins_from_collection(ns, coll, 'modules', spec)
-                except LookupError:
-                    continue
+        if isinstance(el[key], Mapping):
+            for ns in spec.keys():
+                for coll in get_rewritable_collections(ns, spec):
+                    if collection == coll:
+                        # https://github.com/ansible-community/collection_migration/issues/156
+                        continue
 
-                for module in modules_in_collection:
-                    if key != module:
+                    try:
+                        modules_in_collection = get_plugins_from_collection(ns, coll, 'modules', spec)
+                    except LookupError:
                         continue
-                    new_module_name = get_plugin_fqcn(ns, coll, key)
-                    msg = 'Rewriting to %s' % new_module_name
-                    if args.fail_on_core_rewrite:
-                        raise RuntimeError(msg)
 
-                    logger.debug(msg)
-                    translate.append((new_module_name, key))
-                    integration_tests_add_to_deps((namespace, collection), (ns, coll))
+                    for module in modules_in_collection:
```

Comment:
Wow, why is this not
```suggestion
                    if key not in modules_in_collection:
                        continue
```
?

Reply:
Most likely draft code that hasn't been 'fixed'.

Reply:
Will fix in a separate PR.

Reply:
https://github.com/ansible-community/collection_migration/pull/208.